pkg/shell/cockpit-server.js

```diff
@@ -581,11 +571,15 @@ PageShutdownDialog.prototype = {
         else
             when = "+" + delay;
 
-        this.cockpitd_manager.call('Shutdown', op, when, message, function(error) {
-            $('#shutdown-dialog').modal('hide');
-            if (error && error.name != 'Disconnected')
-                shell.show_unexpected_error(error);
-        });
+        var arg = (op == "shutdown") ? "--poweroff" : "--reboot";
+        cockpit.spawn(["shutdown", arg, when, message])
```

Comment:
`shutdown --reboot now` asks for polkit auth and then reboots.

`shutdown --reboot +0` fails with "Must be root".  This is what we execute from Cockpit for "No Delay".

Still don't know why the test succeeds...